grr/server/grr_response_server/gui/ui/components/client_details/client_details.ng.html

```diff
@@ -0,0 +1,241 @@
+      </tr>
+      <tr *ngIf="client.osInfo.installDate">
+        <td class="description">Install date</td>
+        <td class="value"><timestamp [date]="client.osInfo.installDate"></timestamp></td>
+      </tr>
+    </table>
+
+    <br>
+    <div class="section-title mat-h3">Users</div>
+    <table class="section">
+      <ng-template ngFor let-user [ngForOf]="client.users" let-i="index">
+        <tr *ngIf="user.username">
+          <td class="description">Username</td>
+          <td class="value">{{user.username}}</td>
+        </tr>
+        <tr *ngIf="user.fullName">
+          <td class="description">Full name</td>
+          <td class="value">{{user.fullName}}</td>
+        </tr>
+        <tr *ngIf="user.lastLogon">
+          <td class="description">Last logon</td>
+          <td class="value"><timestamp [date]="user.lastLogon"></timestamp></td>
+        </tr>
+        <tr *ngIf="user.homedir">
+          <td class="description">Home directory</td>
+          <td class="value">{{user.homedir}}</td>
+        </tr>
+        <tr *ngIf="user.uid">
+          <td class="description">UID</td>
+          <td class="value">{{user.uid}}</td>
+        </tr>
+        <tr *ngIf="user.gid">
+          <td class="description">GID</td>
+          <td class="value">{{user.gid}}</td>
+        </tr>
+        <tr *ngIf="user.shell">
+          <td class="description">Shell</td>
+          <td class="value">{{user.shell}}</td>
+        </tr>
+        <br *ngIf="i !== client.users.length - 1">
+      </ng-template>
+      <div class="empty-section" *ngIf="client.users.length === 0">Oops, this client seems to have no users.</div>
+    </table>
+
+    <br>
+    <div class="section-title mat-h3">CPU & Memory</div>
+    <table class="section">
+      <tr *ngIf="client.osInfo.machine">
+        <td class="description">CPU Architecture</td>
+        <td class="value">{{client.osInfo.machine}}</td>
+      </tr>
+      <tr *ngIf="client.memorySize">
+        <td class="description">Memory size</td>
+        <td class="value"><human-readable-size [size]="client.memorySize"></human-readable-size></td>
+      </tr>
+    </table>
+
+    <br>
+    <div class="section-title mat-h3">Network Interfaces</div>
+    <table class="section">
+      <ng-template ngFor let-interface [ngForOf]="client.networkInterfaces" let-i="index">
+        <tr>
+          <td class="description">Interface name</td>
+          <td class="value">{{interface.interfaceName}}</td>
+        </tr>
+        <tr>
+          <td class="description">MAC</td>
+          <td class="value">{{interface.macAddress}}</td>
+        </tr>
+        <tr>
+          <td class="description">IP addresses</td>
+          <td class="value">
+            <div *ngFor="let address of interface.addresses">
+              {{address.ipAddress}}
+            </div>
+          </td>
+        </tr>
+        <br *ngIf="i !== client.networkInterfaces.length - 1">
```

Comment:
Use `last` from https://angular.io/api/common/NgForOf#local-variables instead of doing maths yourself.